keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,914 @@
+
+![Figure to the motivation](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/l3-alignment-motivation.png)
+
+The figure below highlights performance gain when pod placement is aligned to an L3 group. In this exampleTPCC/My-SQL  is the workload deployed. There was a 18% uplift compared to default behavior.  Other workloads might see  higher gains.
+
+![Figure to the results of l3 cache alignment](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/results.png) ![Figure of results](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/chart.png)
+
+### Goals
+
+- Introduce a new CPU Manager policy option that assigns CPU within the same grouping of uncorecache or L3 cache to container scope.  
+- Minimize the number of CPUs assignments from different uncorecache/L3 grouping to pods and container scope.
+
+### Non-Goals
+
+This proposal does not aim to modify CPU assignments for CPU Manager policy set to none.  This does not alter the behavior of existing static policy options such as full-pcpus-only.
+
+## Proposal
+
+- Add a new static policy option to fine-tune of CPU assignments to align by uncorecache/L3 cache grouping.
+
+- Topology struct already contains CPUDetails which is a map of CPUInfo.  CPUInfo knows about NUMA, socket, and core IDs associated with a CPU.  We just need to add a new member called uncorecacheId into the CPUInfo struct that tracks whether the CPU is part of a split L3. Functionality to support this was merged in cadvisor with these pull-requests [cadvisor/pr-2849](https://github.com/google/cadvisor/pull/2849) and [cadvisor/pr-2847](https://github.com/google/cadvisor/pull/2847/)
+
+- Handle enablement of the policy option in pkg/kubelet/cm/cpumanager/policy_options.go and check the validity of the user flag and capability of the system.  If topology does not support split L3, grouping by L3 static policy will be ignored.  
+
+- Modify the "Allocate" static policy to check for the option, align-cpus-by-uncorecache. For platforms where SMT is enabled, align-cpus-by-uncorecache will typically take full cores and also works with the option full-pcpus-only.
+
+- Align-cpus-by-uncorecache will be compatible with the default takeByTopologyNUMAPacked algo with future plans to be compatible with the options DistributeCPUsAcrossNUMAOption and DistributeCPUsAcrossCoresOption.
```

Comment:
nit: please use user-visible names (see https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy-options) reather than internal/code name. Here and in the rest of the document

Reply:
done